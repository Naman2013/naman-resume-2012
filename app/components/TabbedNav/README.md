# Tabbed Nav
Dynamically will create a navigation on a variable number of tabs. *parent will handle state changes*
- This component uses flexbox to fit tabs

## How to use
- pass a tab array of { value, label } and an onTabClick handler that will update the state in the parent
- activeTab prop will show a little arrow on the active element
- theme will allow you to override styles of the container element.

![alt text](https://github.com/adam-p/markdown-here/raw/master/src/common/images/icon48.png "Logo Title Text 1")
