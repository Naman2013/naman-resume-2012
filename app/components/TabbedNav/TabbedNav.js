/***********************************
* V4 Dynamic Tabbed Nav
*
*
*
***********************************/

import React, { Component } from 'react';
import PropTypes from 'prop-types';
import classnames from 'classnames';
import uniqueId from 'lodash/uniqueId';
import { triangleUp } from 'app/styles/variables/iconURLs';
import styles from './TabbedNav.style';

const {
  arrayOf,
  func,
  number,
  shape,
  string,
  oneOfType,
} = PropTypes;

class TabbedNav extends Component {
  static propTypes = {
    tabs: arrayOf(shape({
      label: string,
      value: oneOfType([string, number]),
    })).isRequired,
    onTabClick: func.isRequired,
    activeTabValue: oneOfType([string, number]),
    theme: shape({}),
  }

  static defaultProps = {
    activeTabValue: 0,
    theme: {},
  };

  changeActiveTab = (e) => {
    const { tab } = e.currentTarget.dataset;
    this.props.onTabClick(tab);
  }

  render() {
    const {
      tabs,
      activeTabValue,
      theme,
    } = this.props;
    return (
      <div
        key={uniqueId()}
        className="root component-container"
        style={theme}
      >
        {tabs.map(tile => (
          <button
            key={uniqueId()}
            className="split-nav-item-container"
            data-tab={tile.value}
            onClick={this.changeActiveTab}
          >
            <div className="split-nav-item">{tile.label}</div>
            <img
              src={triangleUp}
              className={classnames('arrow', {
                'is-hidden': activeTabValue != tile.value,
              })}
              alt="selected icon"
            />
          </button>
        ))}
        <style jsx>{styles}</style>
      </div>
    );
  }
}

export default TabbedNav;
