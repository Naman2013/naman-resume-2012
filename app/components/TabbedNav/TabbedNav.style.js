import css from 'styled-jsx/css';
import { primaryFont } from 'app/styles/variables/fonts';
import {
  geyser,
  romance,
} from 'app/styles/variables/colors_tiles_v4';
import { faintShadow } from 'app/styles/variables/shadows';

export default css`
.root {
  align-items: center;
  display: flex;
  flex-direction: row;
  font-size: 11px;
  font-weight: bold;
  justify-content: space-evenly;
  padding: 0;
  text-align: center;
  text-transform: uppercase;
  width: 100%;
  background-color: ${romance};
}

.arrow {
  margin-bottom: -5px;
}

.is-hidden {
  visibility: hidden;
}

.split-nav-item {
  cursor: pointer;
  margin: 0 5px;
  margin-top: 15px;
  font-family: ${primaryFont};
  text-transform: uppercase;
}

.component-container {
  ${faintShadow}
}

.split-nav-item-container {
  border: 1px solid ${geyser};
  flex: 0 50%;
}
`;
