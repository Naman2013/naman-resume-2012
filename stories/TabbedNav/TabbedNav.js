import React from 'react';
import { storiesOf } from '@storybook/react';
import TabbedNav from '../../app/components/TabbedNav';

let activeTab = 0;
const changeActiveTab = (idx) => {
  activeTab = idx;
};

const tabs = [
  {
    label: 'Step 1',
    value: 0,
  },
  {
    label: 'Step 2',
    value: 1,
  },
  {
    label: 'Step 3',
    value: 2,
  },
];

storiesOf('TabbedNav', module)
  .add('Renders clickable tabs', () => (
    <TabbedNav
      tabs={[tabs]}
      activeTab={activeTab}
      onTabClick={changeActiveTab}
    />
  ));
