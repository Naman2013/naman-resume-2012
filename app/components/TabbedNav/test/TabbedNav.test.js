import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import sinon from 'sinon';
import noop from 'lodash/noop';
import JOIN_TABS from './testConfig';
import TabbedNav from '../TabbedNav';

let mockedEvents = new Array(JOIN_TABS.length).fill({});

mockedEvents = mockedEvents.map((evt, i) => ({
  currentTarget: {
    dataset: {
      tab: JOIN_TABS[i].value,
    },
  },
}));

it('it should render a tab menu', () => {
  const onTabClickSpy = sinon.spy();
  const testData = {
    onTabClick: onTabClickSpy,
    activeTabValue: '/join/step1',
    tabs: JOIN_TABS,
  };
  const tree = renderer.create(<TabbedNav {...testData} />);

  expect(tree.toJSON()).toMatchSnapshot();
});

it('it should call changeActiveTab with the correct value when a tab is clicked', () => {
  const onTabClickSpy = sinon.spy();
  const testData = {
    onTabClick: onTabClickSpy,
    activeTabValue: '/join/step1',
    tabs: JOIN_TABS,
  };
  const component = shallow(<TabbedNav {...testData} />);
  const button = component.find('button').first();
  button.simulate('click', mockedEvents[0]);
  expect(onTabClickSpy.getCall(0).args[0]).toBe(JOIN_TABS[0].value);
});

it('it should NOT display a triangle icon if it is NOT the active item', () => {
  const testData = {
    onTabClick: noop,
    activeTabValue: '/join/step2',
    tabs: JOIN_TABS,
  };
  const component = shallow(<TabbedNav {...testData} />);
  const button = component.find('button').first();
  const img = button.find('img').first();
  expect(img.props().className).toContain('is-hidden');
});
