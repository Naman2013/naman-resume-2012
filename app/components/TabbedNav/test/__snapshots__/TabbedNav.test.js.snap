// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it should render a tab menu 1`] = `
<div
  className="jsx-1792741512 root component-container"
  style={Object {}}
>
  <button
    className="jsx-1792741512 split-nav-item-container"
    data-tab="/join/step1"
    onClick={[Function]}
  >
    <div
      className="jsx-1792741512 split-nav-item"
    >
      Step 1
    </div>
    <img
      alt="selected icon"
      className="jsx-1792741512 arrow"
      src="https://vega.slooh.com/assets/v4/common/status_triangle_up.svg"
    />
  </button>
  <button
    className="jsx-1792741512 split-nav-item-container"
    data-tab="/join/step2"
    onClick={[Function]}
  >
    <div
      className="jsx-1792741512 split-nav-item"
    >
      Step 2
    </div>
    <img
      alt="selected icon"
      className="jsx-1792741512 arrow is-hidden"
      src="https://vega.slooh.com/assets/v4/common/status_triangle_up.svg"
    />
  </button>
  <button
    className="jsx-1792741512 split-nav-item-container"
    data-tab="/join/step3"
    onClick={[Function]}
  >
    <div
      className="jsx-1792741512 split-nav-item"
    >
      Step 3
    </div>
    <img
      alt="selected icon"
      className="jsx-1792741512 arrow is-hidden"
      src="https://vega.slooh.com/assets/v4/common/status_triangle_up.svg"
    />
  </button>
</div>
`;
